### PR TITLE
[vtk] update to 9.0.1

### PR DESCRIPTION
**Pull request by Neumann-A**
(also solves the paraview issue)

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

**Comment by NancyLi1013** [comment]
The features for `vtk  `failed on x64-windows:

```
CMake Error at CMake/vtkDownload.cmake:8 (message):
  Attempted to download Module_MomentInvariants when VTK_FORBID_DOWNLOADS is
  ON
Call Stack (most recent call first):
  CMake/vtkModuleRemote.cmake:142 (vtk_download_attempt_check)
  Remote/MomentInvariants.remote.cmake:5 (vtk_fetch_module)
  Remote/CMakeLists.txt:9 (include)


-- Found Git: C:/Program Files/Git/cmd/git.exe (found version "2.20.1.windows.1") 
CMake Error at CMake/vtkModuleRemote.cmake:12 (message):
  Failed to clone repository:
  'https://gitlab.kitware.com/vtk/MomentInvariants.git' Clone error is:
  'fatal: could not create work tree dir
  'F:/12149/vcpkg/buildtrees/vtk/src/ad8774e85e-85de89847a/Remote/MomentInvariants':
  File exists

  '
Call Stack (most recent call first):
  CMake/vtkModuleRemote.cmake:107 (_git_clone)
  CMake/vtkModuleRemote.cmake:157 (_fetch_with_git)
  Remote/MomentInvariants.remote.cmake:5 (vtk_fetch_module)
  Remote/CMakeLists.txt:9 (include)
```

**Comment by Neumann-A** [comment]
@NancyLi1013: The feature `all` is also broke in current master https://github.com/microsoft/vcpkg/pull/9960#issuecomment-639501173. Due to that i added the suggestion here: https://github.com/microsoft/vcpkg/pull/9960#issuecomment-639437159 and it should work now but I don't have time to test it.

**Comment by NancyLi1013** [comment]
@Neumann-A Thanks for your update.
I will test these features again.

**Comment by NancyLi1013** [comment]
The feature `all` is also broken due to the same reason as above.

`cuda`  seems to be broken like this:

```
-- The CUDA compiler identification is unknown
CMake Error at CMakeLists.txt:126 (enable_language):
  No CMAKE_CUDA_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CUDACXX" or the CMake cache entry CMAKE_CUDA_COMPILER to the full
  path to the compiler, or to the compiler name if it is in the PATH.
```
Other features failed due to this:

```
CMake Error at CMake/vtkModule.cmake:959 (message):
  The VTK::ParallelMPI4Py module requires the disabled module VTK::mpi.
Call Stack (most recent call first):
  CMakeLists.txt:270 (vtk_module_scan)
```

```
CMake Error at ThirdParty/hdf5/CMakeLists.txt:26 (message):
  An external MPI-aware HDF5 requires that VTK be built with MPI support as
  well.
```

@Neumann-A 
Could you please help make a simple description about the relationships between these features?
I met the same problems on another PR, see comment https://github.com/microsoft/vcpkg/pull/11399#issuecomment-637921703.
 and as @larshg said here https://github.com/microsoft/vcpkg/pull/11399#issuecomment-645912925.

Only on a clean environment can these features pass every time?

I test these features on the repo that I cloned last week and used `git pull` to update to the latest changes. Then I continued to testing.

File: ports/vtk/portfile.cmake [hunk 1]
```diff
@@ -88,11 +88,10 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO Kitware/VTK
-    REF ab278e87b181e3a02082bea7361fbaa3ddafb3ad # v9.0 
-    SHA512 50a324f55b58bc4415f972f711420e83b41a100b27729266db4541c24bc7d7bcd27d9158ce2588178b9b2e43c20b9695ffe382605f5cde331e8371e213655164
+    REF 96e6fa9b3ff245e4d51d49f23d40e9ad8774e85e # v9.0.1
+    SHA512 0efb1845053b6143e5ee7fa081b8be98f6825262c59051e88b2be02497e23362055067b2f811eff82e93eb194e5a9afd2a12e3878a252eb4011a5dab95127a6f
     HEAD_REF master
     PATCHES
-        6890.diff         # already merged upstream
```
Comment (ras0219-msft) [inline ports/vtk/portfile.cmake:95]: Did you also intend to delete the file?

File: ports/vtk/portfile.cmake [hunk 2]
```diff
@@ -128,6 +127,7 @@ vcpkg_configure_cmake(
         -DVTK_INSTALL_PACKAGE_DIR=share/vtk
         -DVTK_INSTALL_RUNTIME_DIR=bin
         -DVTK_FORBID_DOWNLOADS=ON
+        -DVTK_ENABLE_REMOTE_MODULES=OFF
```
Comment (ras0219-msft) [inline ports/vtk/portfile.cmake:130]: This looks like it is reducing the vtk featureset -- should this be turned into a feature?

Comment (Neumann-A) [reply ports/vtk/portfile.cmake:130]: You were the one adding:
`        -DVTK_FORBID_DOWNLOADS=ON`
so 
`        -DVTK_ENABLE_REMOTE_MODULES=OFF`
is already a requirement to even have a chance of making the `all` feature work. 
Either we allow downloads or we disallow downloads and remote modules. 
Comment of one of the devs to vtk remote modules: https://github.com/microsoft/vcpkg/pull/9960#issuecomment-639525894

Comment (ras0219-msft) [reply ports/vtk/portfile.cmake:130]: Thanks for the link, this LGTM

**Comment by ras0219-msft** [comment]
Thanks for the great work yet again!

**Comment by cenit** [comment]
@ras0219-msft @Neumann-A this PR broke opencv[vtk]

**Comment by Neumann-A** [comment]
@cenit Logs?

**Comment by cenit** [comment]
VTK not found, that's all.
Vtk9 changed everything from the target point of view. It needs OpenCV 4.4 or a patch to use it correctly. It's a pity it went completely unnoticed, but again and again as happens so often for vcpkg, it breaks the experience for downstream users.

**Comment by Neumann-A** [comment]
VTK changed their module system completely and made it finally sane. Before that VTK was just a buggy mess. If you want some features be explicitly tested in vcpkg's CI just add test port for it. See https://github.com/microsoft/vcpkg/tree/master/scripts/test_ports 

I mean I also explicitly deactivated VTK in PCL due to upstream not making a lot of progress. At least OpenCV seem to be updateable to make it work. (Or people need to checkout the old VTK port and use a port-overlay.)

**Pull request merged**